shared/datatypes/state_machine.go

```diff
@@ -280,6 +280,13 @@ func (sm *StateMachine) GetMemory(in *Msg, k string) Memory {
 	return Memory{Key: k, Val: buf, logger: sm.logger}
 }
 
+// DeleteMemory deletes a memory for a given key. It is not an error to delete a
+// key that does not exist.
+func (sm *StateMachine) DeleteMemory(in *Msg, k string) {
+	q := `DELETE FROM states WHERE userid=$1 AND pluginname=$2 AND key=$3`
+	sm.db.Exec(q, in.User.ID, sm.pluginName, k)
```

Comment:
I've added in the extra check. I'm not going to add the test in this checkin for the following reasons:
- I'm _really_ new to go and have no idea what the idiomatic way to test is
- there doesn't seem to be any existing infrastructure for a `DeleteMemory.` test to leverage. It's missing a sql mock driver.